src/sensor/pingsensor.cpp

```diff
@@ -22,8 +24,9 @@ void PingSensor::request(int id)
     }
     qCDebug(PING_PROTOCOL_PINGSENSOR) << "Requesting:" << id;
 
-    ping_message_empty m;
-    m.set_id(id);
+    ping_message m(10);
```

Comment:
Why 10 is being used here ?
Should 10 be the default constructor ?

Reply:
This is to construct an arbitrary pingmessage by hand.
Here we are making our 'empty' pingmessage with a message length of 10.

> Should 10 be the default constructor ?

Maybe. If we add 10 as default argument it will conflict with another constructor, i'm not sure if that one is used. We can/should definitely try to make the pingmessage class more robust to memory access violations.